frontend/src/pages/FundsPullPreApprovals.tsx

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useEffect, useState } from "react";
+import { Approval } from "../interfaces/approval";
+import BackendClient from "../services/backendClient";
+import TestnetWarning from "../components/TestnetWarning";
+import Breadcrumbs from "../components/Breadcrumbs";
+import { Container } from "reactstrap";
+import FundsPullPreApprovalsList from "../components/FundsPullPreApproval/FundsPullPreApprovalsList";
+
+const REFRESH_APPROVALS_INTERVAL = 3000;
+
+function FundsPullPreApprovals() {
+  const [approvals, setApprovals] = useState<Approval[]>([]);
+
+  let refreshApprovals = true;
+
+  useEffect(() => {
+    const fetchApprovals = async () => {
+      try {
+        if (refreshApprovals) {
+          setApprovals(await new BackendClient().getAllFundsPullPreApprovals());
+        }
+        setTimeout(fetchApprovals, REFRESH_APPROVALS_INTERVAL);
```

Comment:
Shouldn't the timeout be inside the condition?

Reply:
@vladkucherov - I copied the implementation from home page, and there the timeout is like that. I'm not sure :|